### DISCUSSION TITLE
Communicating with Service Containers

### DISCUSSION THREAD

**Discussion by garryod** [docker/build-push-action#1020] [Q&A] [answered]
Hi all, I've been trying to run this action with a `Dockerfile` which requires a supporting service (MariaDB instance). I am running the service using the [service containers feature](https://docs.github.com/en/actions/using-containerized-services/about-service-containers) but have thus far been unable to establish a connection between it and the docker build. Thus far I have tried:
1. Setting `network` to `host` and forwarding the service container's port
2. Setting `network` to `bridge` using the `driver-opts` option in `setup-buildx`
3. Setting `network` to the `github_network_*` bridge network (found via `docker network ls --filter name=github_network`) using the `driver-opts` option in `setup-buildx`

The networking model is not entirely clear to me, it seems that GitHub actions automatically puts containers run in the workflow into a shared bridge network, but this magic is not present for `docker build` contexts

Does anyone happen to know how I might go about this?

**Comment by larryro** [comment]
second.

**Comment by garryod** [comment, marked as answer]
For anyone who ends up here in the future, I was able to solve this by:
- Forwarding the appropriate port on the service container
- Setting `driver-opts: network=host` on the `setup-buildx-action`
- Passing a `build-args` into the `build-push-action` with a `localhost:port` address

There are a [few examples of this](https://github.com/search?q=language%3AYAML++path%3A%2F%5E.github%5C%2Fworkflows%5C%2F%2F+docker%2Fbuild-push-action+services+&type=code) kicking about, including the one I borrowed from in [nocobase/nocobase](https://github.com/nocobase/nocobase/blob/536001eaba778ce20ff937e548aa113e00190294/.github/workflows/build-docker-image.yml)

**Reply by wgwz** [reply]
You, my friend, are the best.

**Comment by larryro** [comment]
> For anyone who ends up here in the future, I was able to solve this by:
> 
> * Forwarding the appropriate port on the service container
> * Setting `driver-opts: network=host` on the `setup-buildx-action`
> * Passing a `build-args` into the `build-push-action` with a `localhost:port` address
> 
> There are a [few examples of this](https://github.com/search?q=language%3AYAML++path%3A%2F%5E.github%5C%2Fworkflows%5C%2F%2F+docker%2Fbuild-push-action+services+&type=code) kicking about, including the one I borrowed from in [nocobase/nocobase](https://github.com/nocobase/nocobase/blob/536001eaba778ce20ff937e548aa113e00190294/.github/workflows/build-docker-image.yml)

It seems that the key point  is to set `driver-opts: network=host` on `setup-buildx-action`. After completing this setup, I am able to access the port mapped by the service container through `localhost`.

The setting in `build-args` should not be necessary.
